colossalai/shardformer/layer/flash_attention/flash_attention_opt.py

```diff
@@ -0,0 +1,139 @@
+import math
+from copy import deepcopy
+from typing import Callable, List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+from torch.distributed import ProcessGroup
+
+from colossalai.nn import init as init
+from colossalai.shardformer.layer.parallel_module import ParallelModule
+from colossalai.shardformer.layer.utils import create_randomizer_with_offset
+
+__all__ = ['FlashAttentionForOPT']
+
+
+class FlashAttentionForOPT(ParallelModule):
+
+    def __init__(self,
+                 embed_dim: int,
+                 num_heads: int,
+                 dropout: float = 0.0,
+                 is_decoder: bool = False,
+                 bias: bool = True):
+        super().__init__()
+        self.embed_dim = embed_dim
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.head_dim = embed_dim // num_heads
```

Comment:
I notice that this is not a parallel module in fact. A `ParallelModule` must implement tensor parallelism, but this is a normal pytorch module. If this we only need to use xformer in the forward function and everything else is the same as the original native module, we can just choose to replace the forward method?